src/overlays/actors/ovl_En_River_Sound/z_en_river_sound.c

```diff
@@ -27,15 +30,212 @@ const ActorInit En_River_Sound_InitVars = {
+        vec.z = point[-1].z;
+        sp78[0] = func_80AE6A54(&vec, &pointLoc, pos, &sp54);
+    }
+    if (pointIdx + 1 != numPoints) {
+        vec.x = point[1].x;
+        vec.y = point[1].y;
+        vec.z = point[1].z;
+        sp78[1] = func_80AE6A54(&pointLoc, &vec, pos, &sp60);
+    }
+
+    if (sp78[0] && sp78[1]) {
+        if (!func_80AE6A54(&sp54, &sp60, pos, res)) {
+            res->x = (sp54.x + sp60.x) * .5f;
+            res->y = (sp54.y + sp60.y) * .5f;
+            res->z = (sp54.z + sp60.z) * .5f;
+        }
+    } else if (sp78[0]) {
+        res->x = sp54.x;
+        res->y = sp54.y;
+        res->z = sp54.z;
+    } else if (sp78[1]) {
+        res->x = sp60.x;
+        res->y = sp60.y;
+        res->z = sp60.z;
+    } else {
+        res->x = pointLoc.x;
+        res->y = pointLoc.y;
+        res->z = pointLoc.z;
+    }
+
+    return 1;
+}
+
+void EnRiverSound_Update(Actor* thisx, GlobalContext* globalCtx) {
+    Path* path;
+    Vec3f* pos;
+    Player* player = PLAYER;
+    EnRiverSound* this = THIS;
+    s32 sp34;
+
+    if (this->actor.params == 0 || this->actor.params == 4 || this->actor.params == 5) {
+        path = &globalCtx->setupPathList[this->pathIndex];
+        pos = &thisx->world.pos;
+
+        if (func_80AE6BC0(SEGMENTED_TO_VIRTUAL(path->points), path->count, &player->actor.world.pos, pos)) {
+            if (BgCheck_EntityRaycastFloor4(&globalCtx->colCtx, &this->actor.floorPoly, &sp34, &this->actor, pos) !=
+                -32000.f)
+                this->unk_14D = SurfaceType_GetConveyorSpeed(&globalCtx->colCtx, this->actor.floorPoly, sp34);
+            else
+                this->unk_14D = 0;
+
+            if (this->unk_14D == 0) {
+                if (this->actor.params == 4)
+                    this->unk_14D = 0;
+                else if (this->actor.params == 0)
+                    this->unk_14D = 1;
+                else
+                    this->unk_14D = 2;
+            } else {
+                this->unk_14D--;
+                this->unk_14D = CLAMP_MAX(this->unk_14D, 2);
+            }
+        }
+    } else if (this->actor.params == 13 || this->actor.params == 19)
+        func_8002DBD0(&player->actor, &this->actor.home.pos, &this->actor.world.pos);
+    else if (globalCtx->sceneNum == SCENE_DDAN_BOSS && Flags_GetClear(globalCtx, this->actor.room))
+        Actor_Kill(&this->actor);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_River_Sound/func_80AE6BC0.s")
+void EnRiverSound_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnRiverSound* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_River_Sound/EnRiverSound_Update.s")
+    static s16 D_80AE71F8[] = {
+        0x0000, 0x2028, 0x200B, 0x2007, 0x0000, 0x0000, 0x2032, 0x203B, 0x2030, 0x2041, 0x205D,
+        0x0000, 0x20A6, 0x0000, 0x20C0, 0x20CA, 0x20D6, 0x20D7, 0x2098, 0x0000, 0x2031, 0x20E5,
+    };
+    static f32 D_80AE7224[] = {
+        0.7f,
+        1.0f,
+        1.4f,
+    };
```

Comment:
since this array is short, i would take the trailing comma of the last entry. this will make it so the formatter keeps it on one line